src/inet/UDPEndPointImplNetworkFramework.mm

```diff
@@ -0,0 +1,655 @@
+
+constexpr const char * kListenerStateInvalid = "Listener: Invalid";
+constexpr const char * kListenerStateWaiting = "Listener: Waiting";
+constexpr const char * kListenerStateFailed = "Listener: Failed";
+constexpr const char * kListenerStateReady = "Listener: Ready";
+constexpr const char * kListenerStateCancelled = "Listener: Cancelled";
+
+constexpr const char * kConnectionStateInvalid = "Connection: Invalid";
+constexpr const char * kConnectionStateWaiting = "Connection: Waiting";
+constexpr const char * kConnectionStatePreparing = "Connection: Preparing";
+constexpr const char * kConnectionStateFailed = "Connection: Failed";
+constexpr const char * kConnectionStateReady = "Connection: Ready";
+constexpr const char * kConnectionStateCancelled = "Connection: Cancelled";
+
+void DebugPrintConnectionState(nw_connection_state_t state)
+{
+    const char * str = nullptr;
+
+    switch (state) {
+    case nw_connection_state_invalid:
+        str = kConnectionStateInvalid;
+        break;
+    case nw_connection_state_preparing:
+        str = kConnectionStatePreparing;
+        break;
+    case nw_connection_state_waiting:
+        str = kConnectionStateWaiting;
+        break;
+    case nw_connection_state_failed:
+        str = kConnectionStateFailed;
+        break;
+    case nw_connection_state_ready:
+        str = kConnectionStateReady;
+        break;
+    case nw_connection_state_cancelled:
+        str = kConnectionStateCancelled;
+        break;
+    default:
+        chipDie();
+    }
+
+    ChipLogDetail(Inet, "%s", str);
+}
+
+void DebugPrintListenerState(nw_listener_state_t state)
+{
+    const char * str = nullptr;
+
+    switch (state) {
+    case nw_listener_state_invalid:
+        str = kListenerStateInvalid;
+        break;
+    case nw_listener_state_waiting:
+        str = kListenerStateWaiting;
+        break;
+    case nw_listener_state_failed:
+        str = kListenerStateFailed;
+        break;
+    case nw_listener_state_ready:
+        str = kListenerStateReady;
+        break;
+    case nw_listener_state_cancelled:
+        str = kListenerStateCancelled;
+        break;
+    default:
+        chipDie();
+    }
+
+    ChipLogDetail(Inet, "%s", str);
+}
+
+void DebugPrintConnectionPathStatus(nw_path_t path)
+{
+    const char * str = nullptr;
+
+    __auto_type status = nw_path_get_status(path);
+    switch (status) {
+    case nw_path_status_invalid:
+        str = kPathStatusInvalid;
+        break;
+    case nw_path_status_unsatisfied:
+        str = kPathStatusUnsatisfied;
+        break;
+    case nw_path_status_satisfied:
+        str = kPathStatusSatisfied;
+        break;
+    case nw_path_status_satisfiable:
+        str = kPathStatusSatisfiable;
+        break;
+    default:
+        chipDie();
+    }
+
+    ChipLogError(Inet, "%s", str);
+}
+
+void DebugPrintConnection(const nw_connection_t & aConnection)
```

Comment:
Why are we passing `nw_connection_t` by const reference if it's just a pointer?  I'd think we would pass it by value...